src/Kmd.Logic.Gateway.Automation/ValidatePublishing.cs

```diff
@@ -52,7 +52,12 @@ public async Task<ValidationResult> ValidateAsync(string folderPath)
             }
 
             using var client = this.gatewayClientFactory.CreateClient();
-            using var validatePublishingRequest = await GetValidatePublishingRequest(folderPath, this.options.ProviderId, publishFileModel).ConfigureAwait(false);
+            var providerId = this.options.ProviderId.HasValue
```

Comment:
The best place to populate ProviderId is in the constructor of the GatewayAutomation class. That will be a single point of change and almost all others can be avoided(except for some minor ones like making nullable, optional etc).